commit 3e3ae40f053b22fbb9bef50067d6edad4c358c4c
Author: Mirko Vogt <dev@nanl.de>
Date:   Tue May 24 14:36:42 2011 +0200

    use 'fputws_unlocked(S,F)' instead of 'fputws(S,F)'
    
    this eliminates a source of reproduceable freezes

--- a/libc/stdio/_vfprintf.c
+++ b/libc/stdio/_vfprintf.c
@@ -1229,7 +1229,7 @@ static size_t _fp_out_narrow(FILE *fp, i
 #define STRLEN  wcslen
 #define _PPFS_init _ppwfs_init
 /* Pulls in fseek: */
-#define OUTPUT(F,S)			fputws(S,F)
+#define OUTPUT(F,S)			fputws_unlocked(S,F)
 /* TODO: #define OUTPUT(F,S)		_wstdio_fwrite((S),wcslen(S),(F)) */
 #define _outnwcs(stream, wstring, len)	_wstdio_fwrite((const wchar_t *)(wstring), len, stream)
 #define FP_OUT _fp_out_wide
